src/Passes/docs/src/UserGuide/BuildingLibrary.md

```diff
@@ -0,0 +1,149 @@
+
+### On Ubuntu 20.04
+
+Additionally to CMake, you will also require Clang in order to build Ubuntu 20.04. You can accomplish this by following the steps below:
+
+```sh
+apt install clang-11 cmake
+```
+
+You can skip straight to [Library Dependencies](#library-dependencies) if you do not intend to work on developing the core library.
+
+## Developer prerequisites
+
+Development of the core library requires additional tools. These are used formatting, linting and managing code quality. For this purpose, the following tools and packages are used:
+
+- Python 3
+- Python packages specified in `requirements.txt`
+- clang-format
+- clang-tidy
+
+The installation process varies depending on the platform you use. The following subsections provide details on how to install these tools on each platform.
+
+### On OS X
+
+TODO:
+
+### On Ubuntu 20.04
+
+Installing the clang tools on Ubuntu along with Python can be accomplished by running these commands:
+
+```sh
+apt install clang-format-11 clang-tidy-11
+apt install python3 python3-pip
+```
+
+We recommend that you use version 11 of clang in order to be consistent with the version of LLVM on which the library depends. In general, the code should work with any version of clang. Make sure that the compiler environment variables are defined correctly in order to select the correct version:
+
+```sh
+export CC=clang-11
+export CXX=clang++-11
+```
+
+### Common to all platforms
+
+Last but not least, we install the Python libraries that are required:
+
+```sh
+      pip install -r requirements.txt
+      chmod +x manage
+```
+
+As a result, the `manage` tool will be available for you to help you make sure your code is high quality. Getting a contribution merged into the code base requires the manage tools CI flow to be successful. Thus, if you plan to contribute you will most likely need to complete this step.
+
+## Library Dependencies
+
+The QAT library itself is written in C++ and depends on LLVM for compilation and Google Test for testing purposes. Google test is checked out as a submodule whereas LLVM needs to be installed on the system. To get check out all submodules run
+
+```sh
+git submodule update --init --recursive
+```
+
+### Installing LLVM on OS X
+
+TODO:
+
+### Installing LLVM on Ubuntu 20.04
+
+For the installation of LLVM from Ubuntu we use the package manager `apt`. For full run installation, run the following command:
+
+```sh
+apt install  llvm-11 lldb-11 llvm-11-dev libllvm11 llvm-11-runtime
+```
+
+By doing this, we will ensure that CMake can find LLVM and that all headers and libraries used by the QAT library are available.
+
+# Build instructions for users
+
+First
+
+```sh
+cd Debug
+Cmake ..
```

Comment:
> Cmake ..

I suspect you meant 
either `CMake` (the first 2 letters are capital)
or `cmake` (no capital letters)
here.

For now I have an impression that the commad `Cmake` will not be found on the case-sensitive file systems.

Reply:
Fixed - thanks .